install-habitat/cmd/root.go

```diff
@@ -0,0 +1,123 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"runtime"
+
+	"github.com/chef/ci-studio-common/lib"
+
+	"github.com/juju/fslock"
+	"github.com/spf13/cobra"
+)
+
+var (
+	rootCmd = &cobra.Command{
+		Use:   "install-habitat",
+		Short: "Install VERSION of Chef Habitat from CHANNEL.",
+		Run:   maybeInstallHabitat,
+	}
+
+	rootOpts = struct {
+		version string
+		channel string
+		target  string
+	}{}
+
+	globalHabValue = "0.82.0"
+)
+
+func init() {
+	defaultVersion := lib.SettingWithDefault("hab-version", globalHabValue)
+	defaultTarget := lib.SettingWithDefault("hab-target", fmt.Sprintf("x86_64-%s", runtime.GOOS))
+
+	rootCmd.Flags().StringVarP(&rootOpts.version, "version", "v", defaultVersion, "Which version of Habitat you wish to install.")
+	rootCmd.Flags().StringVarP(&rootOpts.channel, "channel", "c", "stable", "The channel from which you wish to install Habitat.")
+	rootCmd.Flags().StringVarP(&rootOpts.target, "target", "t", defaultTarget, "The kernel target for this installation.")
+}
+
+// Execute runs the command
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(-1)
+	}
+}
+
+func maybeInstallHabitat(cmd *cobra.Command, args []string) {
```

Comment:
Should we care about timeouts, we could introduce a `context.Context` here, like
```go
ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
defer cancel()
```
and pass `ctx` along to the different API calls. This way, we'd have a guarantee this wouldn't run longer than 3 minutes; without relying on (unknown to me) default timeouts.